### ISSUE TITLE
[Validator] Default Validator group not validated

### ISSUE THREAD

**Issue by pinoniq** [symfony/symfony#16070]
Symfony framework edition 2.7

I have the following Doctrine entity:

```php
<?php

namespace AppBundle\Doctrine\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Validator\GroupSequenceProviderInterface;

/**
 * Class Article
 * @package AppBundle\Doctrine\Entity
 *
 * @ORM\Entity()
 * @ORM\HasLifecycleCallbacks()
 * @Assert\GroupSequenceProvider
 */
class Article implements GroupSequenceProviderInterface
{
    /**
     * @var string
     *
     * @ORM\Column(type="text")
     * @Assert\NotBlank
     */
    private $content;

    /**
     * @var string
     *
     * @Assert\NotBlank(groups={"Promoted"})
     * @Assert\Length(max="128", groups={"Promoted"})
     *
     * @ORM\Column(name="promotional_title", type="string", length=128, nullable=true)
     */
    private $promotionalTitle;

    /**
     * Adds the promoted group
     * {@inheritdoc}
     */
    public function getGroupSequence()
    {
        $groups = array();
        if ($this->isPromoted()) {
            $groups[] = 'Promoted';
        }
        return $groups;
    }

}
```

This leads to SQL errors when $content is empty. The $content property is never validated. 

Same happens if I set the following:
```
@Assert\NotBlank(groups={'Default'})
...
$groups=array('Default');
```

As in the documentation, the Validator should always validate the Asserts on the $content variable. this is however not the case here.

**Comment by xelaris** [comment]
Thank you for reporting this issue @pinoniq. I think the validator works as intended. The [validation group related documentation] (http://symfony.com/doc/current/book/validation.html#validation-groups) stats:

> As you have already seen in the previous section, the Default group and the group containing the class name (e.g. User) were identical. However, when using Group Sequences, they are no longer identical. The Default group will now reference the group sequence, instead of all constraints that do not belong to any group.

> This means that you have to use the {ClassName} (e.g. User) group when specifying a group sequence.

I guess, if you initialize the `$groups` array with the group named like your class `Article`, as in the [docs code example](http://symfony.com/doc/current/book/validation.html#group-sequence-providers), the validation will work as exptected.

> As in the documentation, the Validator should always validate the Asserts on the $content variable.

Maybe there is an issue in the documentation. Where is this declared?

Status: Works for me

**Comment by pinoniq** [comment]
I had to re-read the docs a couple of times. I think it isn't clear at all about this. I added the {ClassName} so my groupSequence always returns an array with this group and all works.

I think the docs should be clearer about this that when using groupSequences, and not returning the {ClassName} simply doesn't validate the Asserts without any group assigned to them.

If I have time Ill try some rewording myself.
